site/pt-br/guide/effective_tf2.md

```diff
@@ -0,0 +1,287 @@
+
+### O fim das variáveis globais
+
+O TensorFlow 1.X dependia muito de namespaces implicitamente globais. Quando você chamava `tf.Variable()`, ele era colocado no grafo padrão e permanecia lá, mesmo se você perdesse a variável do Python que apontava para ele. Depois, você poderia recuperar esse `tf.Variable`, mas somente se soubesse o nome com que foi criado originalmente. Isto era difícil de fazer se você não tivesse controle sobre a criação da variável. Como resultado, houve uma proliferação de mecanismos para tentar ajudar os usuários a encontrar suas variáveis novamente e para os frameworks encontrarem variáveis criadas por usuários: escopos de variáveis, coleções globais, métodos auxiliares, como `tf.get_global_step()` e `tf.global_variables_initializer()`, otimizadores computando gradientes implicitamente sobre todas as variáveis treináveis e assim por diante. O TensorFlow 2.0 elimina todos esses mecanismos ([RFC Variables 2.0](https://github.com/tensorflow/community/pull/11)), dando lugar ao mecanismo padrão: manter o controle de suas variáveis! Se você perder o controle de uma `tf.Variable`, ela é recolhida pelo coletor de lixo.
+
+A necessidade de rastrear variáveis cria algum trabalho adicional para o usuário, mas, com objetos do Keras (veja abaixo), esse trabalho é minimizado.
+
+### Funções, não sessões
+
+Uma chamada a `session.run()` é quase igual a uma chamada de função: você especifica as entradas e a função a ser chamada, e recebe de volta um conjunto de saídas. No TensorFlow 2.0, você pode decorar uma função do Python usando `tf.function()` que irá marcá-la para a compilação JIT, de forma que seja executada como um único grafo pelo TensorFlow ([RFC Functions 2.0](https://github.com/tensorflow/community/pull/20)). Este mecanismo permite que o TensorFlow 2.0 ganhe todas as vantagens do modo de grafo:
+
+- Desempenho: a função pode ser otimizada (poda de nós, fusão de kernels, etc.)
+- Portabilidade: a função pode ser exportada/reimportada ([RFC SavedModel 2.0](https://github.com/tensorflow/community/pull/34)), permitindo que os usuários reusem e compartilhem funções modulares do TensorFlow.
+
+```python
+# TensorFlow 1.X
+outputs = session.run(f(placeholder), feed_dict={placeholder: input})
+# TensorFlow 2.0
+outputs = f(input)
+```
+
+Com o poder de interpor livremente código do Python e do TensorFlow, os usuários podem tirar proveito da expressividade do Python. Mas o TensorFlow portátil executa em contextos que não possuem um interpretador de Python, como em mobile, C++ e JavaScript. Para ajudar os usuários a evitar ter que reescrever seu código ao adicionar `@tf.function`, o [AutoGraph](function.ipynb) converte um subconjunto de constructs do Python em equivalentes do TensorFlow:
```

Comment:
s/constructs do/código/

Reply:
Could be "... converte um subconjunto de estruturas de código do Python ..."